.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,65 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  run_unit_tests:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '8.x'
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build solution
+      run: dotnet build
+
+    - name: Run tests
+      run: |
+        dotnet test Chefster.Tests/Chefster.Tests.csproj
```

Comment:
Will it stop the deploy if unit tests fail? I assume it probably will

Reply:
yeah it won't run the 'deploy_to_ec2' job if the 'run_unit_test' job fails.
peep line 33

Reply:
ope my bad, nice 👍🏻